File: Server/ChatServer/src/main/java/com/szlazakm/chatserver/dtos/EncryptedMessageDTO.java
package com.szlazakm.chatserver.dtos;

import lombok.*;
import lombok.experimental.SuperBuilder;
import lombok.extern.jackson.Jacksonized;

import java.time.Instant;
import java.util.UUID;

@SuperBuilder
@Jacksonized
@Data
@ToString
public class EncryptedMessageDTO {
    boolean initial;
    UUID id;
    String from;
    String to;
    String cipher;
    String aliceIdentityPublicKey;
    String aliceEphemeralPublicKey;
    Integer bobOpkId;
    Integer bobSpkId;
    String ephemeralRatchetKey;
    int messageIndex;
    int lastMessageBatchSize;
    String phoneNumber;
    Long nonceTimestamp;
    byte[] nonce;
    byte[] authMessageSignature;
}

File: client/app/src/main/java/com/szlazakm/safechat/webclient/dtos/EncryptedMessageDTO.kt
package com.szlazakm.safechat.webclient.dtos

import java.time.Instant

data class EncryptedMessageDTO (
    val initial: Boolean,
    val from: String,
    val to: String,
    val cipher: String,
    val aliceIdentityPublicKey: String?,
    val aliceEphemeralPublicKey: String?,
    val bobOpkId: Int?,
    val bobSpkId: Int?,
    val ephemeralRatchetKey: String,
    val messageIndex: Int,
    val lastMessageBatchSize: Int,

    val phoneNumber: String,
    val nonceTimestamp: Long,
    val nonce: ByteArray,
    val authMessageSignature: ByteArray
)
